src/VirtualClient/VirtualClient.Actions/ApacheBench/ApacheBenchExecutor.cs

```diff
@@ -0,0 +1,251 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO.Abstractions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using VirtualClient.Common;
+    using VirtualClient.Common.Extensions;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+
+    /// <summary>
+    /// Apache http server benchmarking workload executor
+    /// </summary>
+    public class ApacheBenchExecutor : VirtualClientComponent
+    {
+        private IPackageManager packageManager;
+        private IFileSystem fileSystem;
+        private ProcessManager processManager;
+        private ISystemManagement systemManagement;
+        private IStateManager stateManager;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="dependencies">Provides all of the required dependencies to the Virtual Client component.</param>
+        /// <param name="parameters">
+        /// Parameters defined in the execution profile or supplied to the Virtual Client on the command line.
+        /// </param>
+        public ApacheBenchExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters = null)
+            : base(dependencies, parameters)
+        {
+            this.systemManagement = dependencies.GetService<ISystemManagement>();
+            this.fileSystem = dependencies.GetService<IFileSystem>();
+            this.packageManager = dependencies.GetService<IPackageManager>();
+            this.processManager = this.systemManagement.ProcessManager;
+            this.SystemManagement = dependencies.GetService<ISystemManagement>();
+            this.stateManager = this.systemManagement.StateManager;
+        }
+
+        /// <summary>
+        /// The options passed to Apache Bench.
+        /// </summary>
+        public string Options
```

Comment:
What is this options? I dont see it passed in profile.

Reply:
Not useful, Removed.